version.go

```diff
@@ -49,10 +53,15 @@ func AtLeast(major int64, minor int64) error {
 		if err != nil {
 			return err
 		}
-		if version.Variant != "" {
+	}
+
+	if version.Variant != "" {
```

Comment:
You’re still moving the check ouf of the “if version.Major == 0” clause. Why?

Reply:
i only tested this in my app taht runs as a oneoff, my understanding based on the comments above was that this would skip the early out error if it had already been run and fail on subsequent calls, perhaps i misunderstood?

Reply:
Yeah, some check ought to be outside that clause, because there does need to be a passthrough `return nil` if variant is set -- sway uses completely different set of version numbers.

Reply:
This is what I did yesterday when I ran into it:
```diff
diff --git a/version.go b/version.go
index 1a9b815..f9ef441 100644
--- a/version.go
+++ b/version.go
@@ -36,6 +36,8 @@ func GetVersion() (Version, error) {
 // reply. Access only values which don’t change, e.g. Major, Minor.
 var version Version
 
+var variant bool
+
 // AtLeast returns nil if i3’s major version matches major and i3’s minor
 // version is at least minor or newer. Otherwise, it returns an error message
 // stating i3 is too old.
@@ -49,9 +51,13 @@ func AtLeast(major int64, minor int64) error {
 		if err != nil {
 			return err
 		}
+		if version.Variant != "" {
+			log.Printf("non standard i3 payload variant '%s' detected. Ignoring version check. This is fully unsupported.", version.Variant)
+			variant = true
+			return nil
+		}
 	}
-	if version.Variant != "" {
-		log.Printf("non standard i3 payload variant '%s' detected. Ignoring version check. This is fully unsupported.", version.Variant)
+	if variant {
 		return nil
 	}
 	if version.Major == major && version.Minor >= minor {
```

Reply:
i don't see how what you posted above would only log once

Reply:
@stapelberg lmk what you want me to do here

Reply:
Nevermind, I misread—leaving the `if version.Variant != "" ` check outside is the correct thing to do.